File: solvers.julia/optpose2pt_v2/test_num_optpose2pt_v2.jl
include("../_common/header.jl")
include("inidata_num_optpose2pt_v2.jl")
include("coefs_optpose2pt_v2.jl")
include("red_87x120_optpose2pt_v2.jl")

function test_num_optpose2pt_v2()

    Random.seed!(23)
    N = 10000
    d = 24

    stats = ini_stats("optpose2pt_v2")

    for i in 1:N
        data = inidata_num_optpose2pt_v2()
        C,U = coefs_optpose2pt_v2(data)

        toc = @elapsed S = red_87x120_optpose2pt_v2(C)
        stats.tm = [stats.tm; toc]
        if isempty(S); continue; end

        stats = update_stats(stats,S,C,U,d)
    end

    disp_stats(stats)

end

test_num_optpose2pt_v2()

File: solvers.python/py_optpose2pt_v2/test_num_optpose2pt_v2.py
from numpy.random import seed
from time import time
from _common import ini_stats, update_stats, disp_stats
from py_optpose2pt_v2 import inidata_num_optpose2pt_v2, coefs_optpose2pt_v2, red_87x120_optpose2pt_v2

def test_num_optpose2pt_v2():
    seed(23)
    N = 1000
    d = 24

    stats = ini_stats('optpose2pt_v2')

    for i in range(N):
        data = inidata_num_optpose2pt_v2()
        C,U = coefs_optpose2pt_v2(data)

        tic = time()
        S = red_87x120_optpose2pt_v2(C)
        stats['tm'].append(time()-tic)
        if S is None:
            continue

        stats = update_stats(stats,S,C,U,d)

    disp_stats(stats)